### PR TITLE
[CH] Update doc of ClickHouse

### PR DESCRIPTION
**Pull request by zhanglistar**
## What changes were proposed in this pull request?

Update doc of ClickHouse.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: ep/build-clickhouse/src/build_clickhouse.sh [hunk 1]
```diff
@@ -16,5 +23,5 @@ done
 
 echo ${GLUTEN_SOURCE}
 
-cmake -G Ninja -S ${GLUTEN_SOURCE}/cpp-ch -B ${GLUTEN_SOURCE}/cpp-ch/build_ch -DCH_SOURCE_DIR=${GLUTEN_SOURCE}/cpp-ch/ClickHouse "-DCMAKE_C_COMPILER=$(command -v clang-15)" "-DCMAKE_CXX_COMPILER=$(command -v clang++-15)" "-DCMAKE_BUILD_TYPE=Release"
+cmake -G Ninja -S ${GLUTEN_SOURCE}/cpp-ch -B ${GLUTEN_SOURCE}/cpp-ch/build_ch -DCH_SOURCE_DIR=${CH_SOURCE_DIR} "-DCMAKE_C_COMPILER=$(command -v clang-15)" "-DCMAKE_CXX_COMPILER=$(command -v clang++-15)" "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
```
Comment (baibaichen) [inline ep/build-clickhouse/src/build_clickhouse.sh:26]: `RelWithDebInfo` is fine, but `libc.so` is too large, i am not sure how it affects CI.

@liuneng1994  any idea?

Comment (liuneng1994) [reply ep/build-clickhouse/src/build_clickhouse.sh:26]: ci doesn't use the script

Comment (baibaichen) [reply ep/build-clickhouse/src/build_clickhouse.sh:26]: OK

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**